### PR TITLE
ZCS-16234 : Multi-lingual support for password expiry reminder mail

### PR DESCRIPTION
**Pull request by viditZim**
https://synacor.atlassian.net/browse/ZCS-16234

Problem:- Provide multi-lingual support for zimbra password expiry reminder mail

Solution:- Added the translation to the properties files of locales supported in zimbra.

### CODE REVIEW

File: store-conf/conf/msgs/ZsMsg_ja.properties [hunk 1]
```diff
@@ -1000,3 +1000,19 @@ sendPasswordResetEmailBodyHtml =\
...
+  \u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u5909\u66F4\u3055\u308C\u308B\u307E\u305F\u671F\u9650\u5207\u308C\u306B\u306E\u307E\u3057\u305F\u306E\u307E\u3057\u305F\u3002\u000A\u000A\
+  \u3050\u3050\u3050\u3068\u304A\u8A33\u304C\u306A\u306B\u3088\u3063\u3066\u3053\u3046\u3002
+passwordExpiryNotifierBodyHtml=\
+  <p>\u8A73\u3057\u304F\u306F{0}\u69D8\u3002</p>\
+  <p>\u3042\u306A\u305F\u306E\u30A2\u30AB\u30F3\u30C8\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u306F<strong>{1}\u65E5\u5F8C</strong>\u3001<strong>{2}</strong>\u306B\u671F\u9650\u5207\u308C\u306B\u306A\u308A\u307E\u3057\u305F\u3002</p>\
+  <p>\u30A2\u30AB\u30F3\u30C8\u3078\u306E\u30A2\u30AF\u30C8\u3069\u3053\u3093\u304B\u3001\u671F\u9650\u5F8C\u306B\u30D1\u30B9\u30EF\u30FC\u30C9\u3092\u30EA\u30BB\u30C8\u3059\u308B\u3068\u3044\u3046\u8A2D\u5B9A\u306B\u306A\u308A\u307E\u3057\u305F\u3002</p>\
+  <p>\u30D1\u30B9\u30EF\u30FC\u30C9\u304C\u5909\u66F4\u3055\u308C\u308B\u307E\u305F\u671F\u9650\u5207\u308C\u306B\u306E\u307E\u3057\u305F\u306E\u307E\u3057\u305F\u3002</p>\
+  <p>\u3050\u3050\u3050\u3068\u304A\u8A33\u304C\u306A\u306B\u3088\u3063\u3066\u3053\u3046\u3002</p>
```
Comment (k-kato) [inline store-conf/conf/msgs/ZsMsg_ja.properties]: Please update Japanese translation.
```suggestion
passwordExpiryNotifierSubject = \u3042\u306A\u305F\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u306F{0}\u65E5\u5F8C\u306B\u671F\u9650\u5207\u308C\u306B\u306A\u308A\u307E\u3059
passwordExpiryNotifierBodyText=\
  {0}\u69D8\u000A\u000A\
  \u3042\u306A\u305F\u306E\u30A2\u30AB\u30F3\u30C8\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u306F{1}\u65E5\u5F8C\u3001{2}\u306B\u671F\u9650\u5207\u308C\u306B\u306A\u308A\u307E\u3059\u000A\u000A\
  \u30a2\u30ab\u30a6\u30f3\u30c8\u3078\u306e\u30a2\u30af\u30bb\u30b9\u3092\u7dad\u6301\u3059\u308b\u305f\u3081\u3001\u671f\u9650\u5207\u308c\u306b\u306a\u308b\u524d\u306b\u30d1\u30b9\u30ef\u30fc\u30c9\u3092\u5909\u66f4\u3057\u3066\u304f\u3060\u3055\u3044\u3002\u000A\u000A\
  \u30d1\u30b9\u30ef\u30fc\u30c9\u304c\u5909\u66f4\u3055\u308c\u308b\u307e\u3067\u30011\u65e5\u6bce\u306b\u901a\u77e5\u3092\u53d7\u4fe1\u3057\u307e\u3059\u3002\u000A\u000A\
  \u3053\u308c\u306f\u81ea\u52d5\u901a\u77e5\u30e1\u30fc\u30eb\u3067\u3059\u3002
passwordExpiryNotifierBodyHtml=\
  <p>{0}\u69D8</p>\
  <p>\u3042\u306A\u305F\u306E\u30A2\u30AB\u30F3\u30C8\u306E\u30D1\u30B9\u30EF\u30FC\u30C9\u306F{1}\u65E5\u5F8C\u3001{2}\u306B\u671F\u9650\u5207\u308C\u306B\u306A\u308A\u307E\u3059</p>\
  <p>\u30a2\u30ab\u30a6\u30f3\u30c8\u3078\u306e\u30a2\u30af\u30bb\u30b9\u3092\u7dad\u6301\u3059\u308b\u305f\u3081\u3001\u671f\u9650\u5207\u308c\u306b\u306a\u308b\u524d\u306b\u30d1\u30b9\u30ef\u30fc\u30c9\u3092\u5909\u66f4\u3057\u3066\u304f\u3060\u3055\u3044\u3002</p>\
  <p>\u30d1\u30b9\u30ef\u30fc\u30c9\u304c\u5909\u66f4\u3055\u308c\u308b\u307e\u3067\u30011\u65e5\u6bce\u306b\u901a\u77e5\u3092\u53d7\u4fe1\u3057\u307e\u3059\u3002</p>\
  <p>\u3053\u308c\u306f\u81ea\u52d5\u901a\u77e5\u30e1\u30fc\u30eb\u3067\u3059\u3002</p>
```
memo: current and expect messages
```
あなたのパスワードは{0}日後に期限切れになりました
↓
あなたのパスワードは{0}日後に期限切れになります

詳しくは{0}様。
あなたのアカントのパスワードは{1}日後、{2}に期限切れになりました。
アカントへのアクトどこんか、期限後にパスワードをリセトするという設定になりました。
パスワードが変更されるまた期限切れにのましたのました。
ぐぐぐとお訳がなによってこう。
↓
{0}様
あなたのアカントのパスワードは{1}日後、{2}に期限切れになります。
アカウントへのアクセスを維持するため、期限切れになる前にパスワードを変更してください。
パスワードが変更されるまで、1日毎に通知を受信します。
これは自動通知メールです。
```

Comment (viditZim) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: After changes:-

image

Comment (ykomiyam) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: I almost agree with Kato-san's proposal except three words ("様"-->"さん", "1日毎" --> "毎日", "通知を受信します"--> "通知が届きます").  

My suggestion is 
```
{0}さん
あなたのアカントのパスワードは{1}日後、{2}に期限切れになります。
アカウントへのアクセスを維持するため、期限切れになる前にパスワードを変更してください。
パスワードが変更されるまで、毎日通知が届きます。
これは自動通知メールです。
```

Comment (ykomiyam) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: Additionally, Kato-san's suggestion for the reminder message Subject `あなたのパスワードは{0}日後に期限切れになります` looks much natural and easy to understand.

Comment (k-kato) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: Here is unicode data
```
passwordExpiryNotifierSubject = \u3042\u306a\u305f\u306e\u30d1\u30b9\u30ef\u30fc\u30c9\u306f{0}\u65e5\u5f8c\u306b\u671f\u9650\u5207\u308c\u306b\u306a\u308a\u307e\u3059

passwordExpiryNotifierBodyText=\
  {0}\u3055\u3093\n\n\
  \u3042\u306a\u305f\u306e\u30a2\u30ab\u30f3\u30c8\u306e\u30d1\u30b9\u30ef\u30fc\u30c9\u306f{1}\u65e5\u5f8c\u3001{2}\u306b\u671f\u9650\u5207\u308c\u306b\u306a\u308a\u307e\u3059\u3002\n\n\
  \u30a2\u30ab\u30a6\u30f3\u30c8\u3078\u306e\u30a2\u30af\u30bb\u30b9\u3092\u7dad\u6301\u3059\u308b\u305f\u3081\u3001\u671f\u9650\u5207\u308c\u306b\u306a\u308b\u524d\u306b\u30d1\u30b9\u30ef\u30fc\u30c9\u3092\u5909\u66f4\u3057\u3066\u304f\u3060\u3055\u3044\u3002\n\n\
  \u30d1\u30b9\u30ef\u30fc\u30c9\u304c\u5909\u66f4\u3055\u308c\u308b\u307e\u3067\u3001\u6bce\u65e5\u901a\u77e5\u304c\u5c4a\u304d\u307e\u3059\u3002\n\n\
  \u3053\u308c\u306f\u81ea\u52d5\u901a\u77e5\u30e1\u30fc\u30eb\u3067\u3059\u3002

passwordExpiryNotifierBodyHtml=\
  <p>{0}\u3055\u3093</p>\
  <p>\u3042\u306a\u305f\u306e\u30a2\u30ab\u30f3\u30c8\u306e\u30d1\u30b9\u30ef\u30fc\u30c9\u306f{1}\u65e5\u5f8c\u3001{2}\u306b\u671f\u9650\u5207\u308c\u306b\u306a\u308a\u307e\u3059\u3002</p>\
  <p>\u30a2\u30ab\u30a6\u30f3\u30c8\u3078\u306e\u30a2\u30af\u30bb\u30b9\u3092\u7dad\u6301\u3059\u308b\u305f\u3081\u3001\u671f\u9650\u5207\u308c\u306b\u306a\u308b\u524d\u306b\u30d1\u30b9\u30ef\u30fc\u30c9\u3092\u5909\u66f4\u3057\u3066\u304f\u3060\u3055\u3044\u3002</p>\
  <p>\u30d1\u30b9\u30ef\u30fc\u30c9\u304c\u5909\u66f4\u3055\u308c\u308b\u307e\u3067\u3001\u6bce\u65e5\u901a\u77e5\u304c\u5c4a\u304d\u307e\u3059\u3002</p>\
  <p>\u3053\u308c\u306f\u81ea\u52d5\u901a\u77e5\u30e1\u30fc\u30eb\u3067\u3059\u3002</p>
```

Decoded Japanese messages:
```
passwordExpiryNotifierSubject = あなたのパスワードは{0}日後に期限切れになります

passwordExpiryNotifierBodyText=\
  {0}さん\n\n\
  あなたのアカントのパスワードは{1}日後、{2}に期限切れになります。\n\n\
  アカウントへのアクセスを維持するため、期限切れになる前にパスワードを変更してください。\n\n\
  パスワードが変更されるまで、毎日通知が届きます。\n\n\
  これは自動通知メールです。

passwordExpiryNotifierBodyHtml=\
  <p>{0}さん</p>\
  <p>あなたのアカントのパスワードは{1}日後、{2}に期限切れになります。</p>\
  <p>アカウントへのアクセスを維持するため、期限切れになる前にパスワードを変更してください。</p>\
  <p>パスワードが変更されるまで、毎日通知が届きます。</p>\
  <p>これは自動通知メールです。</p>
```

Comment (viditZim) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: Done.

**Review by k-kato — APPROVED** [review]

File: store-conf/conf/msgs/ZsMsg_ja.properties (hunk 1 above)
Comment (k-kato) [reply store-conf/conf/msgs/ZsMsg_ja.properties]: Thank you for the update, adding `<strong></strong>`. I missed it.

**Review by shrutig0510 — APPROVED** [review]

**Review by Gopal-Moolchandani — APPROVED** [review]

**Review by dasiyogesh — APPROVED** [review]

**Pull request merged**